src/terms/xc.jl

```diff
@@ -47,29 +51,38 @@ function ene_ops(term::TermXc, ψ, occ; ρ, kwargs...)
         E += sum(terms.zk .* ρ.real) * dVol
 
         # Add potential contributions Vρ -2 ∇⋅(Vσ ∇ρ)
-        potential .+= terms.vrho
+        for iσ in 1:n_spin
+            potential[iσ] .+= @view terms.vrho[iσ, :, :, :]
+        end
         if haskey(terms, :vsigma)
-            potential .+= -2 * divergence_real(α -> terms.vsigma .* density.∇ρ_real[α],
-                                               density.basis)
+            @assert term.basis.model.spin_polarization in (:none, :spinless)
+            potential[1] .+=
+                -2 * divergence_real(
+                    α -> @view(terms.vsigma[1, :, :, :]) .* density.∇ρ_real[α],
+                    density.basis,
+                )
         end
     end
     if term.scaling_factor != 1
         E *= term.scaling_factor
-        potential .*= term.scaling_factor
+        potential = [pot .*= term.scaling_factor for pot in potential]
```

Comment:
actually for this kind of things I wonder if we haven't complicated our life too much. If you were to do this in Fortran you'd just declare potential to be N1 x N2 x N3 x nspin and be done with it. Having this array of array makes things more complicated. To keep in mind for the array discussion

Reply:
For the potential I'm not too sure, but definitely I'll switch to something more low-level for the derivatives for the PBE.